batch_writer.go

```diff
@@ -34,6 +34,17 @@ func (w *BatchWriter) WriteRowBatch(batch *RowBatch) error {
 			return nil
 		}
 
+		lastRow := batch.Values()[len(batch.Values())-1]
```

Comment:
batch.Size() is the same as len(batch.Values())? In that case, we're safe doing the -1, because it's guaranteed to be >= 0.

Reply:
correction: because it's guaranteed to be greater than zero (not greater or equal)

Reply:
The [ ... ] is guaranteed to be >= 0, we're not disagreeing here, we're just talking about different things.

Reply:
Yeah, `batch.Size()` above basically does the same thing. I refactored them to actually be same in this method.